packages/asset-swapper/src/utils/market_operation_utils/fills.ts

```diff
@@ -150,6 +163,54 @@ function dexQuotesToPaths(
     return paths;
 }
 
+function rfqtIndicativeQuotesToPath(
+    side: MarketOperation,
+    quotes: RfqtIndicativeQuoteResponse[],
+    ethToOutputRate: BigNumber,
+    fees: { [source: string]: BigNumber },
+): Fill[] {
+    // Create a single path from all orders.
+    let path: Fill[] = [];
+    for (const quote of quotes) {
+        const makerAmount = quote.makerAssetAmount;
+        const takerAmount = quote.takerAssetAmount;
+        const input = side === MarketOperation.Sell ? takerAmount : makerAmount;
+        const output = side === MarketOperation.Sell ? makerAmount : takerAmount;
+        const penalty = 0;
+        const rate = makerAmount.div(takerAmount);
+        const adjustedRate =
+            side === MarketOperation.Sell
+                ? makerAmount.minus(penalty).div(takerAmount)
+                : makerAmount.div(takerAmount.plus(penalty));
+        // Skip orders with rates that are <= 0.
+        if (adjustedRate.lte(0)) {
+            continue;
+        }
+        path.push({
+            input,
+            output,
+            rate,
+            adjustedRate,
+            adjustedOutput: output,
```

Comment:
Is there a reason this isn't a calculated `adjustedOutput`? Basically we adjust the `output` by the cost it takes to fill it. Without it then RFQT quotes are basically free.

Penalty should be the same as a normal native order.

Reply:
If we could just pass in the RFQT orders to `nativeOrdersToPath` like @dorothy-zbornak suggested it feels like that would solve this and make the code a lot simpler.  I'm not 100% confident on the implications here though

Reply:
This is moot now.  This function has gone away.  Now the indicative quotes are transformed into native orders much earlier in the pipeline, before this method is used.

Reply:
See here: https://github.com/0xProject/0x-monorepo/pull/2555/commits/d6d4d29257c4cb0810c280f8edc33939f0e6092e#diff-bdb1465a97651026c09ef607e5de9e6fR323